app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestActivity.java

```diff
@@ -0,0 +1,162 @@
+        return new NewContestPresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_viewpager_container;
+    }
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ButterKnife.bind(this);
+        setupToolbar();
+        setupViewPager(viewPager);
+        presenter.initializeContestBuilder(getIntent());
+    }
+
+    public NewContestContract.Presenter getPresenter() {
+        return presenter;
+    }
+
+    private void setupToolbar() {
+        setSupportActionBar(toolbar);
+        ActionBar actionBar = getSupportActionBar();
+        actionBar.setDisplayHomeAsUpEnabled(true);
+        actionBar.setTitle(R.string.new_contest);
+        actionBar.setHomeAsUpIndicator(android.R.drawable.ic_media_rew);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            setupStatusBarColor();
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private void setupStatusBarColor() {
+        Window window = getWindow();
+        window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        window.setStatusBarColor(getResources().getColor(R.color.colorPrimaryDark));
+    }
+
+    public void setupViewPager(ViewPager viewPager) {
+        tabAdapter = new SlidingTabAdapter(this);
+        tabAdapter.addFragment(new NameContestFragment());
+        tabAdapter.addFragment(new DescribeContestFragment());
+        tabAdapter.addFragment(new CategoriesListFragment());
+        viewPager.setAdapter(tabAdapter);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        menu.clear();
+        getMenuInflater().inflate(R.menu.menu_new_contest, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                presenter.onBackButtonClicked();
+                break;
+            case R.id.action_next:
+                presenter.onNextButtonClicked();
+                break;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public int getNumberOfScreens() {
+        return NUMBER_OF_SCREENS;
+    }
+
+    public void acceptContestName(String contestName) {
+        presenter.setContestName(contestName);
+    }
+
+    @Override
+    public void acceptContestDescription(String contestDescription) {
+        presenter.setContestDescription(contestDescription);
+    }
+
+    @Override
+    public void acceptCategory(Category category) {
+        presenter.addCategory(category);
+    }
+
+    @Override
+    public void onCancelClicked() {
+        startActivity(SplashActivity.getIntent(this)); //fixme - NavUtil or onBackPressed - no ned for new intent.
```

Comment:
I used makeIntent() instead of getIntent, we can decide in one of them and standardize our methods afterwards.